server/src/main/java/org/opensearch/cluster/coordination/JoinHelper.java

```diff
@@ -122,6 +132,16 @@ public class JoinHelper {
 
     private final Supplier<JoinTaskExecutor> joinTaskExecutorGenerator;
     private final Consumer<Boolean> nodeCommissioned;
+    private final NamedWriteableRegistry namedWriteableRegistry;
+    private Cache<Version, BytesReference> serializedStates;
+    public static final Setting<TimeValue> CLUSTER_MANAGER_VALIDATE_JOIN_CACHE_INTERVAL = Setting.timeSetting(
+        "cluster_manager.validate_join.cache_interval",
+        TimeValue.timeValueMillis(30000),
+        TimeValue.timeValueMillis(0),
+        TimeValue.timeValueMillis(60000),
```

Comment:
Another nitpick...probably more readable to use `timeValueSeconds`

Reply:
Ack.